platforms/core-runtime/launcher/src/main/java/org/gradle/tooling/internal/provider/LauncherServices.java

```diff
@@ -214,6 +215,12 @@ BuildSessionActionExecutor createActionExecutor(
     }
 
     static class ToolingBuildTreeScopeServices {
+
+        ProblemStream createProblemStream(StartParameter startParameter, ProblemDiagnosticsFactory problemDiagnosticsFactory) {
+            return startParameter.getWarningMode().shouldDisplayMessages()
+                ? problemDiagnosticsFactory.newUnlimitedStream()
+                : problemDiagnosticsFactory.newStream();
```

Comment:
💅 This reads weird to me.  Seems like the `shouldDisplayMessages()` value is _not_ controlling whether or not to display messages, but rather _how many_ messages to display.

Reply:
Good point, I didn't want to mess with this though it is just moving the logic for this here so I can reuse the `ProblemStream` globally.